designer/client/src/containers/useAnonymousStatistics.tsx

```diff
@@ -1,13 +1,61 @@
 import { useSelector } from "react-redux";
 import { getFeatureSettings } from "../reducers/selectors/settings";
-import { useCallback } from "react";
-import { useInterval } from "./Interval";
+import { useCallback, useEffect, useState } from "react";
 import httpService from "../http/HttpService";
+import { useLocalstorageState } from "rooks";
+import moment from "moment";
 
-const STATISTIC_FETCH_TIME_IN_MINUTES = 60;
+const getCacheExpirationDate = (createdAt: number | null, statisticCacheTime: number) =>
+    moment(createdAt).add(statisticCacheTime, "minute");
+const isCacheExpired = (statisticsCache: { createdAt?: number }, statisticCacheTime: number) => {
+    const cacheExpirationDate = getCacheExpirationDate(statisticsCache?.createdAt, statisticCacheTime);
+    const currentDate = moment();
 
-export function useAnonymousStatistics(statisticFetchTime = STATISTIC_FETCH_TIME_IN_MINUTES) {
+    if (!statisticsCache) {
+        return true;
+    }
+
+    return cacheExpirationDate.isSameOrBefore(currentDate);
+};
+
+export const STATISTICS_CACHE = "NU_STATISTICS_CACHE";
+const STATISTIC_CACHE_TIME_IN_MINUTES = 60;
+
+const useStatisticsCache = (statisticCacheTime: number) => {
+    const [statisticsCache, setStatisticsCache] = useLocalstorageState<{ createdAt?: number }>(STATISTICS_CACHE, null);
+
+    const [statisticsReadyToRefetch, setStatisticsReadyToRefetch] = useState(() => {
+        return isCacheExpired(statisticsCache, statisticCacheTime);
+    });
+
+    useEffect(() => {
+        if (statisticsReadyToRefetch) {
+            setStatisticsCache({ createdAt: moment().valueOf() });
+            setStatisticsReadyToRefetch(false);
+        }
+    }, [setStatisticsCache, statisticsReadyToRefetch]);
+
+    useEffect(() => {
+        const remainingTimeToCacheExpired = moment(getCacheExpirationDate(statisticsCache?.createdAt, statisticCacheTime)).diff(
+            moment(),
+            "milliseconds",
+        );
+
+        const timer = setInterval(() => {
+            if (isCacheExpired(statisticsCache, statisticCacheTime)) {
+                setStatisticsReadyToRefetch(true);
+            }
+        }, remainingTimeToCacheExpired);
+
+        return () => clearInterval(timer);
+    }, [statisticCacheTime, statisticsCache]);
+
+    return { statisticsReadyToRefetch };
```

Comment:
maybe just return boolean instead of object with boolean?

Reply:
Done ✔️